mesonbuild/interpreter.py

```diff
@@ -1580,15 +1581,21 @@ def find_library_method(self, args, kwargs):
         return ExternalLibraryHolder(lib, self.subproject)
 
     @permittedKwargs({})
-    def has_argument_method(self, args, kwargs):
+    def has_argument_method(self, args: typing.Sequence[str], kwargs) -> bool:
         args = mesonlib.stringlistify(args)
         if len(args) != 1:
             raise InterpreterException('has_argument takes exactly one argument.')
         return self.has_multi_arguments_method(args, kwargs)
 
     @permittedKwargs({})
-    def has_multi_arguments_method(self, args, kwargs):
+    def has_multi_arguments_method(self, args: typing.Sequence[str], kwargs: dict):
         args = mesonlib.stringlistify(args)
+        # handle arguments with space like "-warn declarations"
+        for i, arg in enumerate(args):
+            smat = re.match(r'^(\-\w+)\ (\w+)$', arg)
+            if smat is not None:
+                args[i] = smat.group(1)
+                args.insert(i + 1, smat.group(2))
```

Comment:
One of the core design points of Meson has been to _never_ do automatic string splitting on whitespace. This does exactly that and hidden behind the scenes at that. I really don't like this. Having one compiler with silly argument handling breaking our entire abstraction layer is not nice.

Reply:
OK--perhaps I can make this method particular to IntelFortranCompiler class, so that it doesn't bother any other compiler. I think that could be acceptable?

Reply:
It's not really the same thing. If you do this, then you are testing if the compiler supports the argument combination `['-warn', 'args']` but during compilation it use the single argument `-warn args`. Meson will quote that space so it passes through shell expansions.

Reply:
Yes in the end I removed all that, and just left the type-hinting and CI-enhancement parts of this PR. I will just use fortran_std instead.